packages/components/src/List/Toolbar/ColumnChooser/hooks/columnChooserClient.hook.js

```diff
@@ -0,0 +1,20 @@
+import { useState, useEffect } from 'react';
+import { mergedColumnsChooser } from '../service';
+
+function useColumnChooserClient(columns, handlerCustomColumnChooser = () => {}) {
+	const [state, setState] = useState({ columns: columns || [] });
+	function handlerColumnChooser(event, { editedColumns }) {
+		setState({ columns: editedColumns });
+	}
+
+	useEffect(() => {
+		handlerCustomColumnChooser({}, state.columns);
+	}, [state.columns]);
+
+	return {
+		state: Object.freeze({ ...state, columns: mergedColumnsChooser(columns, state.columns) }),
```

Comment:
Feels weird to spread the state as it have only the columns in it no ? Maybe i miss something :)

Reply:
I may had another value at one point. I can delete the spreading.